# A phylogenetic analysis 

## Introduction {-}

For this assignment, we were challenged to learn a new skill in four days, any skill related to data science and life sciences that will help you in the future.

After this course I will be specialising in microbiology, this is also where I see myself in a couple of years. Data science for me is a great opportunity to combine different work areas and keep my work various, but my main focus will be microbiology. Before I start working in the field, I hope to earn a masters degree, so in two years I will probably be busy with my masters. I am not sure what masters I would like to do.

Because I am interested in microbiology, my free assignment will also be in that area. I chose to learn to generate a phylogenetic tree and to analyse its contents about the Filoviridae family. I chose this viral family because I am interested in infectious diseases and two species of these are known to cause quite the infectious diseases.

----

## Plan {-}

To achieve my goal, I had 4 days total planned to work on it. I had divided the days in a couple groups:
<ul><li>Half a day to come up with a plan, write this and search the right data needed.</li>
<li>One day to find everything that I need for my own analysis.</li>
<li>Two days to implement my knowledge from searching the internet and tutorials to the data I will be using and complete generating the tree.</li>
<li>Half a day to finish the writing and details.</li></ul>

Eventually, I gained information that I needed to generate a phylogenetic tree from a lot of searching, some trial and error, but also with help from these resources: 
<ul><li>[Multiple Alignment Object](https://bioconductor.org/packages/devel/bioc/vignettes/Biostrings/inst/doc/MultipleAlignments.pdf)</li>
<li>[ggtree utilities](https://bioconductor.statistik.tu-dortmund.de/packages/3.5/bioc/vignettes/ggtree/inst/doc/ggtreeUtilities.html)</li>
<li>[tree Visualization](https://bioconductor.statistik.tu-dortmund.de/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeVisualization.html#displaying-tree-scale-evolution-distance)</li>
<li>[Building a phylogeny in R](https://brouwern.github.io/lbrb/worked-example-building-a-phylogeny-in-r.html#genetic-distances-of-sequence-in-subset)</li>
<li>[Visualization and annotation of phylogenetic trees: ggtree](https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html)</li></ul>

----

## The Filoviridae family {-}

Viruses of the Filoviridae family, also know as filoviruses, are viruses containing one molecule of single-stranded negative-sense RNA that are enveloped in a fatty membrane. The viruses are zoonotic and it is believed that their reservoir hosts usually are bats [@FilovirusesFiloviridaeViral2021]. The viruses are one of the families known to cause Viral Haemorrhagic Fevers (VHFs) that affects multiple body systems and usually results in heavy bleeding and death [@WhatAreVHFs2021].

There are currently five genera in the family; Ebolavirus, Marburgvirus, Cuevavirus, Dianlovirus and Striavirus. Ebolavirus has six known species and the rest have one [@languonFilovirusDiseaseOutbreaks2019]. In my analysis I left Striavirus out, because there is barely any information or data on.

Ebolavirus and Marburgvirus are the most common to cause (deathly) disease in humans, the other ones also host in non-human primates and pigs [@filoviruses-infection-with.pdf].

----

## The phylogenetic tree {-}

In one of my resources I found a useful function from the {compbio4all} package, but Nathan L Brouwer, the writer of the package, states that if there is any trouble downloading the package the (following) code for the function can be copied. The package in question can be found at [the github repository](https://github.com/brouwern/compbio4all.git). I did have trouble installing the package, so the code retrieved from github is below.

```{r function codes}

# Function to clean up fasta files
fasta_cleaner <- function(fasta_object, parse = TRUE){

  fasta_object <- sub("^(>)(.*?)(\\n)(.*)(\\n\\n)","\\4",fasta_object)
  fasta_object <- gsub("\n", "", fasta_object)

  if(parse == TRUE){
    fasta_object <- stringr::str_split(fasta_object,
                                       pattern = "",
                                       simplify = FALSE)
  }

  return(fasta_object[[1]])
}

# Function to store fasta sequences into a R list
entrez_fetch_list <- function(db, id, rettype, ...){

  #setup list for storing output
  n.seq <- length(id)
  list.output <- as.list(rep(NA, n.seq))
  names(list.output) <- id

  # get output
  for(i in 1:length(id)){
    list.output[[i]] <- rentrez::entrez_fetch(db = db,
                                              id = id[i],
                                              rettype = rettype)
  }


  return(list.output)
}

```

The analysis starts by fetching the sequences from NCBI nucleotide bank [@NationalCenterBiotechnology] and make them compatible for analysis.
The file "species" is written to enhance the target IDs, this way, all the genomes can be fetched at once.

```{r load and adjust sequences}

# Import list of target species
species <-scan(file = ("files/filo_data/species.txt"), what = "character", sep = "\n", comment.char = "#") 

# Obtain fasta sequences of all species combined in a list
filoviridae <- entrez_fetch_list(db = "nucleotide",
             id = species,
             rettype = "fasta") 

# Remove non sequence infromation
for(i in 1:length(filoviridae)){
  filoviridae[[i]] <- fasta_cleaner(filoviridae[[i]], parse = F)
}

# Create string set
filoviridae_vector <- rep(NA, length(filoviridae))

for(i in 1:length(filoviridae_vector)){
  filoviridae_vector[i] <- filoviridae[[i]]
}

names(filoviridae_vector) <- names(filoviridae)

filoviridae_vector_ss <- Biostrings::AAStringSet(filoviridae_vector)

```

Next is the multiple sequence alignment. This creates an alignment to easily spot the distances between all the options. Because this takes a long time to perform, I saved the alignment in a file that can be obtained.

```{r multiple sequence alignment, eval = FALSE}

# Perform multiple sequence alignment
filoviridae_align <- msa(filoviridae_vector_ss,
                     method = "ClustalW")

# Change class into correct one, in this case RNA because filoviruses are RNA viruses
class(filoviridae_align) <- "RNAMultipleAlignment"

# Save in file so alignment only needs to run once
saveRDS(filoviridae_align, ("files/filo_data/filoviridae_align.RDS"))

```

```{r import aligned data}

# Import aligned file for usage
filoviridae_align <- readRDS("files/filo_data/filoviridae_align.RDS")

```

To use the alignment for the tree, the genetic distance between the sequences will be calculated. For readability, a rounded version is also made. After the calculations the tree is plotted. The tree clearly shows that most ebolaviruses are closely related, which is logical because they are only different species, only Reston and Sudan are somewhat further from the rest but close to each other. It also shows that Mengla dianlovirus and Marburg marburgvirus are closely related, and Lloviu is also nearby.

```{r view msa}

# Adjust datatype for calculations
filoviridae_align_seqinr <- msaConvert(filoviridae_align, 
                                   type = "seqinr::alignment")

# Calculate genetic distances
filoviridae_dist <- seqinr::dist.alignment(filoviridae_align_seqinr, 
                                       matrix = "identity")
filoviridae_dist_rounded <- round(filoviridae_dist,
                              digits = 2) # Add rounded version

```

```{r phylogenetic tree}

# Generate the structure of the tree
filoviridae_tree <- nj(filoviridae_dist)

# Plot tree
ggtree(filoviridae_tree, ladderize = T, branch.length = "branch.length", 
       root.position = .05)+ 
  geom_tiplab(size = 3.5)+
  xlim(0, .8)+
  labs(title = "Filoviridae family tree",  
       subtitle = "Rooted phylogenetic tree, with branch lengths")+
  theme(plot.title = element_text(size = 18, face = "bold", hjust = .5), plot.subtitle = element_text(size = 12, face = "italic", hjust = .5))

```

_Figure 3: Phylogenetic tree of the Filoviridae family._

The tree shows an easy picture of the differences, but is doesn't show the numbers in detail. To solve this, I calculated them separately. The response of the code shows the calculated genetic distances of all options.

```{r details}

# Details
filoviridae_dist_rounded
mima(filoviridae_dist_rounded) # 0.56, 0.76

```

Seen from here it states that Taiforest ebolavirus & Bundibogyo ebolavirus are closest related with a score of 0.56.

It also states that there are multiple options that score furthest related with a score of 0.76. These are: Bombali ebolavirus & Marburg marburgvirus, Bombali ebolavirus & Mengla dianlovirus, Sudan ebolavirus & Marburg marburgvirus, Sudan ebolavirus & Mengla dianlovirus, Reston ebolavirus & Marburg marburgvirus, Lloviu cuevavirus & Marburg marburgvirus, Lloviu cuevavirus & Mengla dianlovirus.

----

## Future steps {-}

In the future I would like to expand my knowledge in phylogeny and generating trees, because this was just a start, there are so much more options to create. A couple of examples of things I would still like to learn in this area are:
<ul><li>How to add the genetic distances to the tree so it is clear in one view.</li>
<li> How to generate different types of trees, because there are a lot more and for different data, different types of trees are more compatible.</li>
<li> How to read an extensive tree. The one I've created is doable, but once they get a lot more values, reading it becomes quite hard.</li></ul>