<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 A phylogenetic analysis | portfolio_book.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.31 and GitBook 2.6.7" />

  <meta property="og:title" content="3 A phylogenetic analysis | portfolio_book.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 A phylogenetic analysis | portfolio_book.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-management.html"/>
<link rel="next" href="group-project.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="form.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<div class="title">Portfolio of Coco Niemel</div>
<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Curriculum Vitae</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#contact-information"><i class="fa fa-check"></i>Contact information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#education"><i class="fa fa-check"></i>Education</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#professional-experience"><i class="fa fa-check"></i>Professional Experience</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#skills"><i class="fa fa-check"></i>Skills</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#languages"><i class="fa fa-check"></i>Languages</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>1</b> Reproducible Research</a>
<ul>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#introduction"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#analyse-an-excel-file"><i class="fa fa-check"></i>Analyse an excel file</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#scoring-reproducibility"><i class="fa fa-check"></i>Scoring reproducibility</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#usage-of-other-peoples-codes"><i class="fa fa-check"></i>Usage of other peoples codes</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-management.html"><a href="data-management.html"><i class="fa fa-check"></i><b>2</b> Data management</a>
<ul>
<li class="chapter" data-level="" data-path="data-management.html"><a href="data-management.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="data-management.html"><a href="data-management.html#my-execution"><i class="fa fa-check"></i>My execution</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html"><i class="fa fa-check"></i><b>3</b> A phylogenetic analysis</a>
<ul>
<li class="chapter" data-level="" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html#introduction-2"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html#plan"><i class="fa fa-check"></i>Plan</a></li>
<li class="chapter" data-level="" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html#the-filoviridae-family"><i class="fa fa-check"></i>The Filoviridae family</a></li>
<li class="chapter" data-level="" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html#the-phylogenetic-tree"><i class="fa fa-check"></i>The phylogenetic tree</a></li>
<li class="chapter" data-level="" data-path="a-phylogenetic-analysis.html"><a href="a-phylogenetic-analysis.html#future-steps"><i class="fa fa-check"></i>Future steps</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="group-project.html"><a href="group-project.html"><i class="fa fa-check"></i><b>4</b> Group project</a>
<ul>
<li class="chapter" data-level="" data-path="group-project.html"><a href="group-project.html#introduction-3"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="group-project.html"><a href="group-project.html#ontox"><i class="fa fa-check"></i>ONTOX</a></li>
<li class="chapter" data-level="" data-path="group-project.html"><a href="group-project.html#our-addition"><i class="fa fa-check"></i>Our addition</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html"><i class="fa fa-check"></i><b>5</b> Relational data and databases</a>
<ul>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#introduction-4"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#dbeaver-connection"><i class="fa fa-check"></i>DBeaver connection</a></li>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#inspection"><i class="fa fa-check"></i>Inspection</a></li>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive statistics</a></li>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#visualisations"><i class="fa fa-check"></i>Visualisations</a>
<ul>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#flu"><i class="fa fa-check"></i>Flu</a></li>
<li class="chapter" data-level="" data-path="relational-data-and-databases.html"><a href="relational-data-and-databases.html#dengue"><i class="fa fa-check"></i>Dengue</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="r-package.html"><a href="r-package.html"><i class="fa fa-check"></i><b>6</b> R package</a>
<ul>
<li class="chapter" data-level="" data-path="r-package.html"><a href="r-package.html#introduction-5"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="r-package.html"><a href="r-package.html#copopa"><i class="fa fa-check"></i>Copopa</a></li>
<li class="chapter" data-level="" data-path="r-package.html"><a href="r-package.html#install-and-load"><i class="fa fa-check"></i>Install and load</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rmarkdown-parameters.html"><a href="rmarkdown-parameters.html"><i class="fa fa-check"></i><b>7</b> RMarkdown parameters</a>
<ul>
<li class="chapter" data-level="" data-path="rmarkdown-parameters.html"><a href="rmarkdown-parameters.html#introduction-6"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="rmarkdown-parameters.html"><a href="rmarkdown-parameters.html#data-inspection"><i class="fa fa-check"></i>Data inspection</a></li>
<li class="chapter" data-level="" data-path="rmarkdown-parameters.html"><a href="rmarkdown-parameters.html#building-a-parameterized-graph"><i class="fa fa-check"></i>Building a parameterized graph</a></li>
<li class="chapter" data-level="" data-path="rmarkdown-parameters.html"><a href="rmarkdown-parameters.html#alter-parameter-settings"><i class="fa fa-check"></i>Alter parameter settings</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a-phylogenetic-analysis" class="section level1" number="3">
<h1><span class="header-section-number">3</span> A phylogenetic analysis</h1>
<div id="introduction-2" class="section level2 unnumbered">
<h2>Introduction</h2>
<p>For this assignment, we were challenged to learn a new skill in four days, any skill related to data science and life sciences that will help you in the future.</p>
<p>After this course I will be specializing in microbiology, this is also where I see myself in a couple of years. Data science for me is a great opportunity to combine different work areas and keep my work various, but my main focus will be microbiology. Before i start working in the field, I hope to earn a masters degree, so in two years i will probably be busy with my masters. I am not sure what masters I would like to do.</p>
<p>Because I am interested in microbiology, my free assignment will also be in that lane. I chose to learn to generate a phylogenetic tree and to analyse its contents about the Filoviridae family. I chose this viral family because I am interested in infectious diseases and two species of these are known to cause quite the infectious diseases.</p>
<hr />
</div>
<div id="plan" class="section level2 unnumbered">
<h2>Plan</h2>
To achieve my goal, I had 4 days total planned to work on it. I had divided the days in a couple groups:
<ul>
<li>
Half a day to come up with a plan, write this and search the right data needed.
</li>
<li>
One day to find everything that I need for my own analysis.
</li>
<li>
Two days to implement my knowledge from searching the internet and tutorials to the data I will be using and complete generating the tree.
</li>
<li>
Half a day to finish the writing and details.
</li>
</ul>
Eventually, i gained my needed information to generate a phylogenetic tree from a lot of searching, some trial and error, but also with help from these resources:
<ul>
<li>
<a href="https://bioconductor.org/packages/devel/bioc/vignettes/Biostrings/inst/doc/MultipleAlignments.pdf">Multiple Alignment Object</a>
</li>
<li>
<a href="https://bioconductor.statistik.tu-dortmund.de/packages/3.5/bioc/vignettes/ggtree/inst/doc/ggtreeUtilities.html">ggtree utilities</a>
</li>
<li>
<a href="https://bioconductor.statistik.tu-dortmund.de/packages/3.8/bioc/vignettes/ggtree/inst/doc/treeVisualization.html#displaying-tree-scale-evolution-distance">tree Visualization</a>
</li>
<li>
<a href="https://brouwern.github.io/lbrb/worked-example-building-a-phylogeny-in-r.html#genetic-distances-of-sequence-in-subset">Building a phylogeny in R</a>
</li>
<li>
<a href="https://guangchuangyu.github.io/ggtree-book/chapter-ggtree.html">Visualization and annotation of phylogenetic trees: ggtree</a>
</li>
</ul>
<hr />
</div>
<div id="the-filoviridae-family" class="section level2 unnumbered">
<h2>The Filoviridae family</h2>
<p>Viruses of the Filoviridae family, also know as filoviruses, are viruses containing one molecule of single-stranded negative-sense RNA that are enveloped in a fatty membrane. The viruses are zoonotic and it is believed that their reservoir hosts usually are bats <span class="citation">(<a href="references.html#ref-FilovirusesFiloviridaeViral2021" role="doc-biblioref"><span>“Filoviruses (<span>Filoviridae</span>) | <span>Viral Hemorrhagic Fevers</span> (<span>VHFs</span>) | <span>CDC</span>”</span> 2021</a>)</span>. The viruses are one of the families known to cause Viral Haemorrhagic Fevers (VHFs) that affects multiple body systems and usually results in heavy bleeding and death <span class="citation">(<a href="references.html#ref-WhatAreVHFs2021" role="doc-biblioref"><span>“What Are <span>VHFs</span>? | <span>Viral Hemorrhagic Fevers</span> (<span>VHFs</span>) | <span>CDC</span>”</span> 2021</a>)</span>.</p>
<p>There are currently five genera in the family; Ebolavirus, Marburgvirus, Cuevavirus, Dianlovirus and Striavirus. Ebolavirus has six knows species and the rest have one <span class="citation">(<a href="references.html#ref-languonFilovirusDiseaseOutbreaks2019" role="doc-biblioref">Languon and Quaye 2019</a>)</span>. In my analysis I left Striavirus out, because there is barely any information or data about.</p>
<p>Ebolavirus and Marburgvirus are the most common to cause (deathly) disease in humans, the other ones also host in non-human primates and pigs <span class="citation">(<a href="#ref-filoviruses-infection-with.pdf" role="doc-biblioref"><strong>filoviruses-infection-with.pdf?</strong></a>)</span>.</p>
<hr />
</div>
<div id="the-phylogenetic-tree" class="section level2 unnumbered">
<h2>The phylogenetic tree</h2>
<p>In one of my resources I found a useful function from the {compbio4all} package, but Nathan L Brouwer, the writer of the package, states that if there is any trouble downloading the package the (following) code for the function can be copied. The package in question can be found at <a href="https://github.com/brouwern/compbio4all.git">the github repository</a>. I did have trouble installing the package, so the code retrieved from github is below.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="a-phylogenetic-analysis.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to clean up fasta files</span></span>
<span id="cb13-2"><a href="a-phylogenetic-analysis.html#cb13-2" aria-hidden="true" tabindex="-1"></a>fasta_cleaner <span class="ot">&lt;-</span> <span class="cf">function</span>(fasta_object, <span class="at">parse =</span> <span class="cn">TRUE</span>){</span>
<span id="cb13-3"><a href="a-phylogenetic-analysis.html#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="a-phylogenetic-analysis.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  fasta_object <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;^(&gt;)(.*?)(</span><span class="sc">\\</span><span class="st">n)(.*)(</span><span class="sc">\\</span><span class="st">n</span><span class="sc">\\</span><span class="st">n)&quot;</span>,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">4&quot;</span>,fasta_object)</span>
<span id="cb13-5"><a href="a-phylogenetic-analysis.html#cb13-5" aria-hidden="true" tabindex="-1"></a>  fasta_object <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="st">&quot;&quot;</span>, fasta_object)</span>
<span id="cb13-6"><a href="a-phylogenetic-analysis.html#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="a-phylogenetic-analysis.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(parse <span class="sc">==</span> <span class="cn">TRUE</span>){</span>
<span id="cb13-8"><a href="a-phylogenetic-analysis.html#cb13-8" aria-hidden="true" tabindex="-1"></a>    fasta_object <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_split</span>(fasta_object,</span>
<span id="cb13-9"><a href="a-phylogenetic-analysis.html#cb13-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">pattern =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb13-10"><a href="a-phylogenetic-analysis.html#cb13-10" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-11"><a href="a-phylogenetic-analysis.html#cb13-11" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-12"><a href="a-phylogenetic-analysis.html#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="a-phylogenetic-analysis.html#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(fasta_object[[<span class="dv">1</span>]])</span>
<span id="cb13-14"><a href="a-phylogenetic-analysis.html#cb13-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-15"><a href="a-phylogenetic-analysis.html#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="a-phylogenetic-analysis.html#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to store fasta sequences into a R list</span></span>
<span id="cb13-17"><a href="a-phylogenetic-analysis.html#cb13-17" aria-hidden="true" tabindex="-1"></a>entrez_fetch_list <span class="ot">&lt;-</span> <span class="cf">function</span>(db, id, rettype, ...){</span>
<span id="cb13-18"><a href="a-phylogenetic-analysis.html#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="a-phylogenetic-analysis.html#cb13-19" aria-hidden="true" tabindex="-1"></a>  <span class="co">#setup list for storing output</span></span>
<span id="cb13-20"><a href="a-phylogenetic-analysis.html#cb13-20" aria-hidden="true" tabindex="-1"></a>  n.seq <span class="ot">&lt;-</span> <span class="fu">length</span>(id)</span>
<span id="cb13-21"><a href="a-phylogenetic-analysis.html#cb13-21" aria-hidden="true" tabindex="-1"></a>  list.output <span class="ot">&lt;-</span> <span class="fu">as.list</span>(<span class="fu">rep</span>(<span class="cn">NA</span>, n.seq))</span>
<span id="cb13-22"><a href="a-phylogenetic-analysis.html#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(list.output) <span class="ot">&lt;-</span> id</span>
<span id="cb13-23"><a href="a-phylogenetic-analysis.html#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="a-phylogenetic-analysis.html#cb13-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># get output</span></span>
<span id="cb13-25"><a href="a-phylogenetic-analysis.html#cb13-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(id)){</span>
<span id="cb13-26"><a href="a-phylogenetic-analysis.html#cb13-26" aria-hidden="true" tabindex="-1"></a>    list.output[[i]] <span class="ot">&lt;-</span> rentrez<span class="sc">::</span><span class="fu">entrez_fetch</span>(<span class="at">db =</span> db,</span>
<span id="cb13-27"><a href="a-phylogenetic-analysis.html#cb13-27" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">id =</span> id[i],</span>
<span id="cb13-28"><a href="a-phylogenetic-analysis.html#cb13-28" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">rettype =</span> rettype)</span>
<span id="cb13-29"><a href="a-phylogenetic-analysis.html#cb13-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-30"><a href="a-phylogenetic-analysis.html#cb13-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-31"><a href="a-phylogenetic-analysis.html#cb13-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-32"><a href="a-phylogenetic-analysis.html#cb13-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(list.output)</span>
<span id="cb13-33"><a href="a-phylogenetic-analysis.html#cb13-33" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>The analysis starts by fetching the sequences from NCBI <span class="citation">(<a href="references.html#ref-NationalCenterBiotechnology" role="doc-biblioref"><span>“National <span>Center</span> for <span>Biotechnology Information</span>,”</span> n.d.</a>)</span> and make them compatible for analysis.
The file “species” is written to enhance the target IDs, this way, all the genomes can be fetched at once.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="a-phylogenetic-analysis.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import list of target species</span></span>
<span id="cb14-2"><a href="a-phylogenetic-analysis.html#cb14-2" aria-hidden="true" tabindex="-1"></a>species <span class="ot">&lt;-</span><span class="fu">scan</span>(<span class="at">file =</span> (<span class="st">&quot;files/filo_data/species.txt&quot;</span>), <span class="at">what =</span> <span class="st">&quot;character&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">comment.char =</span> <span class="st">&quot;#&quot;</span>) </span>
<span id="cb14-3"><a href="a-phylogenetic-analysis.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="a-phylogenetic-analysis.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain fasta sequences of all species combined in a list</span></span>
<span id="cb14-5"><a href="a-phylogenetic-analysis.html#cb14-5" aria-hidden="true" tabindex="-1"></a>filoviridae <span class="ot">&lt;-</span> <span class="fu">entrez_fetch_list</span>(<span class="at">db =</span> <span class="st">&quot;nucleotide&quot;</span>,</span>
<span id="cb14-6"><a href="a-phylogenetic-analysis.html#cb14-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">id =</span> species,</span>
<span id="cb14-7"><a href="a-phylogenetic-analysis.html#cb14-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">rettype =</span> <span class="st">&quot;fasta&quot;</span>) </span>
<span id="cb14-8"><a href="a-phylogenetic-analysis.html#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="a-phylogenetic-analysis.html#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove non sequence infromation</span></span>
<span id="cb14-10"><a href="a-phylogenetic-analysis.html#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(filoviridae)){</span>
<span id="cb14-11"><a href="a-phylogenetic-analysis.html#cb14-11" aria-hidden="true" tabindex="-1"></a>  filoviridae[[i]] <span class="ot">&lt;-</span> <span class="fu">fasta_cleaner</span>(filoviridae[[i]], <span class="at">parse =</span> F)</span>
<span id="cb14-12"><a href="a-phylogenetic-analysis.html#cb14-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-13"><a href="a-phylogenetic-analysis.html#cb14-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-14"><a href="a-phylogenetic-analysis.html#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create string set</span></span>
<span id="cb14-15"><a href="a-phylogenetic-analysis.html#cb14-15" aria-hidden="true" tabindex="-1"></a>filoviridae_vector <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(filoviridae))</span>
<span id="cb14-16"><a href="a-phylogenetic-analysis.html#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="a-phylogenetic-analysis.html#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(filoviridae_vector)){</span>
<span id="cb14-18"><a href="a-phylogenetic-analysis.html#cb14-18" aria-hidden="true" tabindex="-1"></a>  filoviridae_vector[i] <span class="ot">&lt;-</span> filoviridae[[i]]</span>
<span id="cb14-19"><a href="a-phylogenetic-analysis.html#cb14-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-20"><a href="a-phylogenetic-analysis.html#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="a-phylogenetic-analysis.html#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(filoviridae_vector) <span class="ot">&lt;-</span> <span class="fu">names</span>(filoviridae)</span>
<span id="cb14-22"><a href="a-phylogenetic-analysis.html#cb14-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-23"><a href="a-phylogenetic-analysis.html#cb14-23" aria-hidden="true" tabindex="-1"></a>filoviridae_vector_ss <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">AAStringSet</span>(filoviridae_vector)</span></code></pre></div>
<p>Next is the multiple sequence alignment. This creates an alignment to easily spot the distances between all the options. Because this takes a long time to perform, I saved the alignment in a file that can be obtained.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="a-phylogenetic-analysis.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform multiple sequence alignment</span></span>
<span id="cb15-2"><a href="a-phylogenetic-analysis.html#cb15-2" aria-hidden="true" tabindex="-1"></a>filoviridae_align <span class="ot">&lt;-</span> <span class="fu">msa</span>(filoviridae_vector_ss,</span>
<span id="cb15-3"><a href="a-phylogenetic-analysis.html#cb15-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;ClustalW&quot;</span>)</span>
<span id="cb15-4"><a href="a-phylogenetic-analysis.html#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="a-phylogenetic-analysis.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Change class into correct one, in this case RNA because filoviruses are RNA viruses</span></span>
<span id="cb15-6"><a href="a-phylogenetic-analysis.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(filoviridae_align) <span class="ot">&lt;-</span> <span class="st">&quot;RNAMultipleAlignment&quot;</span></span>
<span id="cb15-7"><a href="a-phylogenetic-analysis.html#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="a-phylogenetic-analysis.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Save in file so alignment only needs to run once</span></span>
<span id="cb15-9"><a href="a-phylogenetic-analysis.html#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(filoviridae_align, (<span class="st">&quot;files/filo_data/filoviridae_align.RDS&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="a-phylogenetic-analysis.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import aligned file for usage</span></span>
<span id="cb16-2"><a href="a-phylogenetic-analysis.html#cb16-2" aria-hidden="true" tabindex="-1"></a>filoviridae_align <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;files/filo_data/filoviridae_align.RDS&quot;</span>)</span></code></pre></div>
<p>To use the alignment for the tree, the genetic distance between the sequences will be calculated. For readability, a rounded version is also made. After the calculations the tree is plotted. The tree clearly shows that most ebolaviruses are closely related, which is logical because they are only different species, only Reston and Sudan are somewhat further from the rest but close to each other. Is also shows that Mengla dianlovirus and Marburg marburgvirus are closely related, and Lloviu is also nearby.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="a-phylogenetic-analysis.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust datatype for calculations</span></span>
<span id="cb17-2"><a href="a-phylogenetic-analysis.html#cb17-2" aria-hidden="true" tabindex="-1"></a>filoviridae_align_seqinr <span class="ot">&lt;-</span> <span class="fu">msaConvert</span>(filoviridae_align, </span>
<span id="cb17-3"><a href="a-phylogenetic-analysis.html#cb17-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">type =</span> <span class="st">&quot;seqinr::alignment&quot;</span>)</span>
<span id="cb17-4"><a href="a-phylogenetic-analysis.html#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="a-phylogenetic-analysis.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate genetic distances</span></span>
<span id="cb17-6"><a href="a-phylogenetic-analysis.html#cb17-6" aria-hidden="true" tabindex="-1"></a>filoviridae_dist <span class="ot">&lt;-</span> seqinr<span class="sc">::</span><span class="fu">dist.alignment</span>(filoviridae_align_seqinr, </span>
<span id="cb17-7"><a href="a-phylogenetic-analysis.html#cb17-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">matrix =</span> <span class="st">&quot;identity&quot;</span>)</span>
<span id="cb17-8"><a href="a-phylogenetic-analysis.html#cb17-8" aria-hidden="true" tabindex="-1"></a>filoviridae_dist_rounded <span class="ot">&lt;-</span> <span class="fu">round</span>(filoviridae_dist,</span>
<span id="cb17-9"><a href="a-phylogenetic-analysis.html#cb17-9" aria-hidden="true" tabindex="-1"></a>                              <span class="at">digits =</span> <span class="dv">2</span>) <span class="co"># Add rounded version</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="a-phylogenetic-analysis.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate the structure of the tree</span></span>
<span id="cb18-2"><a href="a-phylogenetic-analysis.html#cb18-2" aria-hidden="true" tabindex="-1"></a>filoviridae_tree <span class="ot">&lt;-</span> <span class="fu">nj</span>(filoviridae_dist)</span>
<span id="cb18-3"><a href="a-phylogenetic-analysis.html#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="a-phylogenetic-analysis.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot tree</span></span>
<span id="cb18-5"><a href="a-phylogenetic-analysis.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggtree</span>(filoviridae_tree, <span class="at">ladderize =</span> T, <span class="at">branch.length =</span> <span class="st">&quot;branch.length&quot;</span>, </span>
<span id="cb18-6"><a href="a-phylogenetic-analysis.html#cb18-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">root.position =</span> .<span class="dv">05</span>)<span class="sc">+</span> </span>
<span id="cb18-7"><a href="a-phylogenetic-analysis.html#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="at">size =</span> <span class="fl">3.5</span>)<span class="sc">+</span></span>
<span id="cb18-8"><a href="a-phylogenetic-analysis.html#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, .<span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb18-9"><a href="a-phylogenetic-analysis.html#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Filoviridae family tree&quot;</span>,  </span>
<span id="cb18-10"><a href="a-phylogenetic-analysis.html#cb18-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">&quot;Rooted phylogenetic tree, with branch lengths&quot;</span>)<span class="sc">+</span></span>
<span id="cb18-11"><a href="a-phylogenetic-analysis.html#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">hjust =</span> .<span class="dv">5</span>), <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;italic&quot;</span>, <span class="at">hjust =</span> .<span class="dv">5</span>))</span></code></pre></div>
<p><img src="portfolio_book_files/figure-html/phylogenetic%20tree-1.png" width="672" /></p>
<p><em>Figure 3: Phylogenetic tree of the Filoviridae family.</em></p>
<p>The tree shows an easy picture of the differences, but is doesn’t show the numbers in detail. To solve this, I calculated them separately.</p>
<details>
<summary>
The response of the code shows the calculated genetic distances of all options.
</summary>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="a-phylogenetic-analysis.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Details</span></span>
<span id="cb19-2"><a href="a-phylogenetic-analysis.html#cb19-2" aria-hidden="true" tabindex="-1"></a>filoviridae_dist_rounded</span></code></pre></div>
<pre><code>##                                 KC545393.1 Bundibogyo ebolavirus
## MH121167.1 Taiforest ebolavirus                             0.56
## KT357858.1 Zaire ebolavirus                                 0.61
## NC_039345.1 Bombali ebolavirus                              0.64
## EU338380.1 Sudan ebolavirus                                 0.65
## MF540571.1 Reston ebolavirus                                0.65
## MZ541881.1 Lloviu cuevavirus                                0.70
## OL702894.1 Marburg marburgvirus                             0.75
## NC_055510.1 Mengla dianlovirus                              0.75
##                                 MH121167.1 Taiforest ebolavirus
## MH121167.1 Taiforest ebolavirus                                
## KT357858.1 Zaire ebolavirus                                0.61
## NC_039345.1 Bombali ebolavirus                             0.64
## EU338380.1 Sudan ebolavirus                                0.65
## MF540571.1 Reston ebolavirus                               0.65
## MZ541881.1 Lloviu cuevavirus                               0.70
## OL702894.1 Marburg marburgvirus                            0.76
## NC_055510.1 Mengla dianlovirus                             0.75
##                                 KT357858.1 Zaire ebolavirus
## MH121167.1 Taiforest ebolavirus                            
## KT357858.1 Zaire ebolavirus                                
## NC_039345.1 Bombali ebolavirus                         0.63
## EU338380.1 Sudan ebolavirus                            0.65
## MF540571.1 Reston ebolavirus                           0.65
## MZ541881.1 Lloviu cuevavirus                           0.71
## OL702894.1 Marburg marburgvirus                        0.75
## NC_055510.1 Mengla dianlovirus                         0.75
##                                 NC_039345.1 Bombali ebolavirus
## MH121167.1 Taiforest ebolavirus                               
## KT357858.1 Zaire ebolavirus                                   
## NC_039345.1 Bombali ebolavirus                                
## EU338380.1 Sudan ebolavirus                               0.66
## MF540571.1 Reston ebolavirus                              0.66
## MZ541881.1 Lloviu cuevavirus                              0.70
## OL702894.1 Marburg marburgvirus                           0.76
## NC_055510.1 Mengla dianlovirus                            0.76
##                                 EU338380.1 Sudan ebolavirus
## MH121167.1 Taiforest ebolavirus                            
## KT357858.1 Zaire ebolavirus                                
## NC_039345.1 Bombali ebolavirus                             
## EU338380.1 Sudan ebolavirus                                
## MF540571.1 Reston ebolavirus                           0.63
## MZ541881.1 Lloviu cuevavirus                           0.71
## OL702894.1 Marburg marburgvirus                        0.76
## NC_055510.1 Mengla dianlovirus                         0.76
##                                 MF540571.1 Reston ebolavirus
## MH121167.1 Taiforest ebolavirus                             
## KT357858.1 Zaire ebolavirus                                 
## NC_039345.1 Bombali ebolavirus                              
## EU338380.1 Sudan ebolavirus                                 
## MF540571.1 Reston ebolavirus                                
## MZ541881.1 Lloviu cuevavirus                            0.71
## OL702894.1 Marburg marburgvirus                         0.76
## NC_055510.1 Mengla dianlovirus                          0.75
##                                 MZ541881.1 Lloviu cuevavirus
## MH121167.1 Taiforest ebolavirus                             
## KT357858.1 Zaire ebolavirus                                 
## NC_039345.1 Bombali ebolavirus                              
## EU338380.1 Sudan ebolavirus                                 
## MF540571.1 Reston ebolavirus                                
## MZ541881.1 Lloviu cuevavirus                                
## OL702894.1 Marburg marburgvirus                         0.76
## NC_055510.1 Mengla dianlovirus                          0.76
##                                 OL702894.1 Marburg marburgvirus
## MH121167.1 Taiforest ebolavirus                                
## KT357858.1 Zaire ebolavirus                                    
## NC_039345.1 Bombali ebolavirus                                 
## EU338380.1 Sudan ebolavirus                                    
## MF540571.1 Reston ebolavirus                                   
## MZ541881.1 Lloviu cuevavirus                                   
## OL702894.1 Marburg marburgvirus                                
## NC_055510.1 Mengla dianlovirus                             0.67</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="a-phylogenetic-analysis.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mima</span>(filoviridae_dist_rounded) <span class="co"># 0.56, 0.76</span></span></code></pre></div>
<pre><code>## $min
## [1] 0.56
## 
## $max
## [1] 0.76</code></pre>
</details>
<p>Seen from here is states that Taiforest ebolavirus &amp; Bundibogyo ebolavirus are closest related with a score of 0.56.</p>
<p>It also states that there are multiple options that score furthest related with a score of 0.76. These are: Bombali ebolavirus &amp; Marburg marburgvirus, Bombali ebolavirus &amp; Mengla dianlovirus, Sudan ebolavirus &amp; Marburg marburgvirus, Sudan ebolavirus &amp; Mengla dianlovirus, Reston ebolavirus &amp; Marburg marburgvirus, Lloviu cuevavirus &amp; Marburg marburgvirus, Lloviu cuevavirus &amp; Mengla dianlovirus.</p>
<hr />
</div>
<div id="future-steps" class="section level2 unnumbered">
<h2>Future steps</h2>
In the future I would like to expand my knowledge in phylogeny and generating trees, because this was just a start, there are so much more options to create. A couple of examples of things I would still like to learn in this area are:
<ul>
<li>
How to add the genetic distances to the tree so it is clear in one view.
</li>
<li>
How to generate different types of trees, because there are a lot more and for different data, different types of trees are more compatible.
</li>
<li>
How to read an extensive tree. The one I’ve created is doable, but once they get a lot more values, reading it becomes quite hard.
</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-management.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="group-project.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
